harvester/pmh_interface.py

```diff
@@ -16,11 +16,18 @@ def __init__(self, url):
         """
         self.sickle = Sickle(url)
 
-    def get_all_metadata_records(self):
+    def get_all_metadata_records(self, limit=None):
         """
         Get content for all records available from the API.
+
+        :param limit: Maximum number of records to retrieve. Default None fetches all records.
         """
-        metadata_records = self.sickle.ListRecords(metadataPrefix="info")
-        for metadata_record in metadata_records:
-            yield metadata_record
+        metadata_records = self.sickle.ListRecords(metadataPrefix="cmdi0571")
+        if limit is None:
```

Comment:
If you want to be pythonic:
```
if not limit:
```
(does not really matter)

Reply:
Also changed this style.